### ISSUE TITLE
Update Neural Network Weights

### ISSUE THREAD

**Issue by ssshathi** [ssshathi/PET-kaggle#2]
The weights of the neural network must be updated via some gradient descent method, SGD for now.
Therefore, backpropagation must be implemented for both log loss and square loss.

This Issue is considered closed when:

- [ ] Gradients for all weights in the network are calculated for Log Loss
- [ ] Gradients for all weights in the network are calculated for Square Loss
- [ ] SGD is used to update weights

If this issue is straightforward to implement, and we want an additional challenge, see #3
